airflow/contrib/hooks/salesforce_hook.py

```diff
@@ -18,127 +18,110 @@
-which allows you to connect to your Salesforce instance,
-retrieve data from it, and write that data to a file
-for other uses.
+This module contains a Salesforce Hook which allows you to connect to your Salesforce instance,
+retrieve data from it, and write that data to a file for other uses.
 
-NOTE:   this hook also relies on the simple_salesforce package:
-        https://github.com/simple-salesforce/simple-salesforce
+.. note:: this hook also relies on the simple_salesforce package:
+      https://github.com/simple-salesforce/simple-salesforce
 """
-from simple_salesforce import Salesforce
-from airflow.hooks.base_hook import BaseHook
-
-import json
+import time
 
 import pandas as pd
-import time
+from simple_salesforce import Salesforce
 
+from airflow.hooks.base_hook import BaseHook
 from airflow.utils.log.logging_mixin import LoggingMixin
 
 
 class SalesforceHook(BaseHook):
-    def __init__(
-            self,
-            conn_id,
-            *args,
-            **kwargs
-    ):
+
+    def __init__(self, conn_id):
         """
-        Create new connection to Salesforce
-        and allows you to pull data out of SFDC and save it to a file.
+        Create new connection to Salesforce and allows you to pull data out of SFDC and save it to a file.
+
+        You can then use that file with other Airflow operators to move the data into another data source.
 
-        You can then use that file with other
-        Airflow operators to move the data into another data source
+        :param conn_id: the name of the connection that has the parameters we need to connect to Salesforce.
+                        The connection should be type `http` and include a user's security token
+                        in the `Extras` field.
+        :type conn_id: str
 
-        :param conn_id:     the name of the connection that has the parameters
-                            we need to connect to Salesforce.
-                            The connection should be type `http` and include a
-                            user's security token in the `Extras` field.
         .. note::
             For the HTTP connection type, you can include a
             JSON structure in the `Extras` field.
             We need a user's security token to connect to Salesforce.
             So we define it in the `Extras` field as:
                 `{"security_token":"YOUR_SECURITY_TOKEN"}`
+
         """
+        super(SalesforceHook, self).__init__(conn_id)
         self.conn_id = conn_id
-        self._args = args
-        self._kwargs = kwargs
+        self.conn = None
 
-        # get the connection parameters
-        self.connection = self.get_connection(conn_id)
-        self.extras = self.connection.extra_dejson
-
-    def sign_in(self):
+    def get_conn(self):
         """
-        Sign into Salesforce.
-
-        If we have already signed it, this will just return the original object
+        Sign into Salesforce, only If we are not already signed in.
```

Comment:
```suggestion
        Sign into Salesforce, only if we are not already signed in.
```
or you just want to point out this?